product_sequence/models/product_product.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2004 Tiny SPRL
+# Copyright 2016 Sodexis
+# Copyright 2018 ForgeFlow S.L.
+#   (http://www.forgeflow.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    default_code = fields.Char(
+        required=True,
+        default="/",
+        tracking=True,
+        help="Set to '/' and save if you want a new internal reference "
+        "to be proposed.",
+    )
+
+    @api.model
+    def create(self, vals):
+        if "default_code" not in vals or vals["default_code"] == "/":
+            categ_id = vals.get("categ_id")
+            template_id = vals.get("product_tmpl_id")
```

Comment:
```suggestion
            categ_id = vals.get("categ_id", False)
            template_id = vals.get("product_tmpl_id", False)
```
Like the above comment.